src/libraries/System.Security.Principal.Windows/src/System/Security/Principal/WindowsIdentity.cs

```diff
@@ -322,19 +322,22 @@ void IDeserializationCallback.OnDeserialization(object sender)
 
         public static WindowsIdentity GetCurrent()
         {
-            return GetCurrentInternal(TokenAccessLevels.MaximumAllowed, false);
+            // not null when threadOnly argument is false
+            return GetCurrentInternal(TokenAccessLevels.MaximumAllowed, false)!;
         }
 
 
-        public static WindowsIdentity GetCurrent(bool ifImpersonating)
+        public static WindowsIdentity? GetCurrent(bool ifImpersonating)
         {
+
             return GetCurrentInternal(TokenAccessLevels.MaximumAllowed, ifImpersonating);
         }
 
 
         public static WindowsIdentity GetCurrent(TokenAccessLevels desiredAccess)
         {
-            return GetCurrentInternal(desiredAccess, false);
+            // not null when threadOnly argument is false
+            return GetCurrentInternal(desiredAccess, false)!;
```

Comment:
It feels like there should be a way to express that "return is non-null when argument is true/false" but i see there isn't.

Reply:
Yeah, something like that would be helpful here.

Reply:
It is one of the attributes we considered and punted on initially (not just for true/false, but for any value that can be determined and compared statically). It shows up in other places, like in reflection methods that take a throw-if-not-found argument. That said, it hasn't come up frequently enough to warrant inclusion yet.
cc: @jcouv, @madstorgersen